### DISCUSSION TITLE
How to use enahnce api for multipe code generated files

### DISCUSSION THREAD

**Discussion by deyancyan** [reduxjs/redux-toolkit#3809] [Q&A] [answered]
I went through the Q&As but did not seem to find exactly what I was looking for. Closest was this https://github.com/reduxjs/redux-toolkit/discussions/2506#discussion-4209201

So my issue is the following - I have a base api file and several code generated files that use **injectEndpoints**. Also what I would like is to be able to use **enhanceEndpoints** for tag management and response transformation. My structure looks like this:
![image](https://github.com/reduxjs/redux-toolkit/assets/116868447/48de33e7-39cf-4eae-9a8a-81937bdc1e7e)

Lets say I do this in the **api-enhanced** file:
![image](https://github.com/reduxjs/redux-toolkit/assets/116868447/045f8db2-867a-4750-99e8-779e15694000)
Now I am using the **enhancedApi** export to configure the store but when I log the order of files execution I get the following:
- api-base
- api-enhanced
- bulk (or any other code generated file that seems to be used on the current page)

This seems to be a problem as the configuration done in **api-enhanced** does not have any effect.

On the other hand when I do the following:
![image](https://github.com/reduxjs/redux-toolkit/assets/116868447/deec09cc-3e05-4456-96a3-bf727187d948)
This seems to correctly work correctly when I use the **enhancedApi** export to configure the store. I am not sure which api definition to use and how to mitigate such behaviour since using the last enhanced api as an export is kind of weird.

Maybe combining all enhanced api's into a single would provide some peace of mind. I get it that they all work on the same reference but it is still quite strange especially for someone else that has no debbuged the issue.

Any ideas?

**Comment by EskiMojo14** [comment, marked as answer]
enhanceEndpoints should be called on the version of the API that has the endpoints you need injected (i.e. the result of calling injectEndpoints). this ensures that they'll be injected *before* the enhancement runs.

**Reply by deyancyan** [reply]
This one I did figure out. Also TS is recognizing the endpoint names this way. But the other question remains - what to do when you have multiple files with **injectEndpoints** and how to configure the store?

Just calling **enhanceEndpoints** on every api chunk is fine but yields no result when I configure the store with the base api.

**Reply by EskiMojo14** [reply]
assuming you don't have control over the generated files:
```ts
// src/api/base
export const baseApi = createApi();

// src/api/bulk
export const bulkApi = baseApi.injectEndpoints();

// if you can control it, this could be in the same file as the original injectEndpoints call
// src/api/bulk-enhanced
export const enhancedBulkApi = bulkApi.enhanceEndpoints();

export const { useABulkQuery } = enhancedBulkApi // by importing this hook somewhere you ensure that the inject and enhance calls happen

// src/store
import { baseApi } from "src/api/base";

export const store = configureStore({
  reducers: {
    [baseApi.reducerPath]: baseApi.reducer
  },
  middleware: getDefaultMiddleware => getDefaultMiddleware().concat(baseApi.middleware)
})
```

**Reply by deyancyan** [reply]
Ah, interesting. I have not tried to export the hooks from the enhancedApi. I was using the ones that are exported during code generation. Will give this a go. Thanks

**Reply by EskiMojo14** [reply]
yeah, ultimately you can only ensure the code happens by importing from it - otherwise the code is never actually executed

**Reply by deyancyan** [reply]
Great, seems to work.

Another problem that I found is that I had **package.json>sideEffects: false** and that is why **import './api-enhanced'** in the index file failed to evaluate the script.

Many thanks!
